File: cpp/72_covering_segments_242B.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    if (!(cin >> n))
        return 0;
    int bestL = INT_MAX, bestR = -1, bestIdx = -1;
    for (int i = 1; i <= n; i++)
    {
        int l, r;
        cin >> l >> r;
        if (l < bestL || (l == bestL && r > bestR))
        {
            bestL = l;
            bestR = r;
            bestIdx = i;
        }
    }
    if (bestIdx == -1)
        cout << "-1\n";
    else
        cout << bestIdx << "\n";
}

File: python/72_covering_segments_242B.py
import sys

lines = sys.stdin.read().strip().split()
if not lines:
    sys.exit(0)
it = iter(lines)
n = int(next(it))
bestL = 10**18
bestR = -1
bestIdx = -1
for i in range(1, n + 1):
    l = int(next(it))
    r = int(next(it))
    if l < bestL or (l == bestL and r > bestR):
        bestL = l
        bestR = r
        bestIdx = i
print(bestIdx if bestIdx != -1 else -1)
